### PR TITLE
[BUG] fix optimizer passed as an instance not being bound to the network in BaseDeepClassifierPytorch

### LINKED ISSUE #10990: [BUG] `optimizer` passed as an instance is not bound to the network
**Describe the bug**

`BaseDeepClassifierPytorch` documents `optimizer` as accepting "an instance of an optimizer defined in PyTorch". When an instance is passed, the estimator fits without error but **the network is never updated** — `predict` returns the output of a randomly initialized network.

A PyTorch optimizer captures the concrete parameter tensors it will update in `param_groups` at construction time. `self.network` is only created inside `_fit`, so at `__init__` time there is no network to bind to. `_instantiate_optimizer` then returns the user's object unchanged:

https://github.com/sktime/sktime/blob/b750673cb9c873e1eaabf3adfb8c8f44280ac0ca/sktime/classification/deep_learning/base/_base_torch.py#L615-L617

`_run_epoch` calls `self._optimizer.step()` , which walks `param_groups` and updates whatever tensors the user happened to bind — not `self.network`. Gradients land correctly on `self.network`'s `.grad` fields and are then discarded. There is no way for a caller to construct a correct instance, since the network does not exist until `fit` runs. 

By contrast the `str` branch is correct, because it constructs the optimizer inside `_fit`, after the network exists:
https://github.com/sktime/sktime/blob/b750673cb9c873e1eaabf3adfb8c8f44280ac0ca/sktime/classification/deep_learning/base/_base_torch.py#L605-L606
This affects every estimator deriving from `BaseDeepClassifierPytorch` and`BaseDeepRegressorPytorch`.

**To Reproduce**

```python
import numpy as np, torch
from sktime.classification.deep_learning.mlp._mlp_torch import MLPClassifierTorch

X = np.random.randn(20, 1, 24)
y = np.array(["a", "b"] * 10)

# the network does not exist yet, so the optimizer must be bound to something else
placeholder = torch.nn.Linear(4, 3)
opt = torch.optim.Adam(placeholder.parameters(), lr=0.01)

print("--- optimizer passed as an instance ---")
MLPClassifierTorch(optimizer=opt, num_epochs=5, verbose=True, random_state=0).fit(X, y)

print("--- equivalent optimizer passed as a string ---")
MLPClassifierTorch(optimizer="adam", lr=0.01, num_epochs=5, verbose=True,
                   random_state=0).fit(X, y)
```

Output:

```
--- optimizer passed as an INSTANCE ---
Epoch 1: Loss: 0.6956636488437653
Epoch 2: Loss: 0.6931753516197204
Epoch 3: Loss: 0.6916144222021103
Epoch 4: Loss: 0.6931950002908707
Epoch 5: Loss: 0.6963465094566346
--- equivalent optimizer passed as a STRING ---
Epoch 1: Loss: 1.6750205263495446
Epoch 2: Loss: 1.0345402874052525
Epoch 3: Loss: 1.0676873428747058
Epoch 4: Loss: 3.1677918057199577
Epoch 5: Loss: 0.9844802880656047
```

The instance run sits at `ln(2) = 0.693` — the loss of an untrained binary classifier. 

**Expected behavior**

Either the optimizer actually updates `self.network`, or passing an instance is rejected with a clear error instead of silently producing an unfitted model.

### PR DESCRIPTION
**Pull request by srupat**
#### Reference Issues/PRs

Fixes #10990  
(Also fixes some minor bugs like mismatch between defaults and docstrings, and some naming conventions that I *prefer* over the existing ones)

#### What does this implement/fix? Explain your changes.

Re-instantiates optimizer with network parameters along with existing parameters. Constructor parameters are given highest preference -- followed by instance parameters (for example, if `lr` has a default value in the constructor, and user has also passed a `lr` value as an optimizer instance parameter, then the parameter value is considered. If `lr` is not passed in the instance, but user has passed a non-default value in the constructor, then the constructor value is considered).

#### Does your contribution introduce a new dependency? If yes, which one?
No. 

#### What should a reviewer concentrate their feedback on?
- Proper docstrings
- `_instantiate_optimizer` method

#### Did you add any tests for the change?
Yes, added tests in `sktime/classification/tests/test_base.py` to test base class changes.

### CODE REVIEW

File: sktime/classification/deep_learning/base/_base_torch.py [hunk 1]
```diff
@@ -595,36 +602,54 @@ def _instantiate_optimizer(self):
             }
         # import the base class for all optimizers in PyTorch
         torchOptimizer = _safe_import("torch.optim.Optimizer")
+
+        # if no optimizer is passed, use Adam as default
+        if not self.optimizer:
```
Comment (fkiraly) [inline sktime/classification/deep_learning/base/_base_torch.py]: explicitly check for None

Comment (fkiraly) [reply sktime/classification/deep_learning/base/_base_torch.py]: @srupat, kindly action this

Comment (srupat) [reply sktime/classification/deep_learning/base/_base_torch.py]: Already fixed in 9049669

File: sktime/classification/deep_learning/base/_base_torch.py [hunk 2]
```diff
@@ -595,36 +602,54 @@ def _instantiate_optimizer(self):
...
-                )
-                if self.optimizer_kwargs:
-                    return optimizer_class(
-                        self.network.parameters(), lr=self.lr, **self.optimizer_kwargs
-                    )
-                else:
-                    return optimizer_class(self.network.parameters(), lr=self.lr)
-            else:
+        elif isinstance(self.optimizer, str):
+            if self.optimizer.lower() not in self._all_optimizers:
```
Comment (fkiraly) [inline sktime/classification/deep_learning/base/_base_torch.py:612]: what we want to do instead, in this case:

* import all objects in `torch.optim`
* create a dict with `x.lower() : x` for all `x` in `torch.optim`
* see if `self.optimizer.lower()` it is in there

Only if not, raise the error.

**Review by fkiraly — APPROVED** [review]
Great!

I think it should also possible to pass strings aliasing optimizers not in the dictionary. But that could be a separate PR; I will open an issue.

File: sktime/classification/deep_learning/base/_base_torch.py (hunk 2 above)
Comment (srupat) [reply sktime/classification/deep_learning/base/_base_torch.py:612]: Alright, so like the activations case? Sure, will pick it up soon.
